### PR TITLE
ci: move from appveyor/travis to github actions

### LINKED ISSUE #93: Infra: Switch to GitHub Actions
- [ ] Replace Travis/Appveyor
- [ ] Set up GH actions for linux and windows builds

### PR DESCRIPTION
**Pull request by cdierkens**
move all CI to github actions while dropping runs for node 10, and adding runs for node 16. Runs now include Windows, MacOS, and Ubuntu on node 12, 14, and 16.  Breaking: Drops node 10 CI unit testing, though package may continue to work. closes #93

## Checklist:

- [x] All tests are passing
- [x] Converts appveyer `yarn run test-ie` to github actions for windows matrix only
- [x] Converts travis `yarn run test` to github actions for all tests
- [x] Update CI badge to point to GH actions

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -0,0 +1,58 @@
+---
+name: Main
+
+on:
+  push:
+    branches:
+      - main
```
Comment (ryan-roemer) [inline .github/workflows/main.yml:7]: Default branch is `master`

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+          yarn install --frozen-lockfile
+          yarn test
+      - name: Test IE
+        if: ${{ matrix.os == 'windows-latest' }}
+        run: |
+          yarn install --frozen-lockfile
+          yarn test-ie
+
+  codecov:
+    if: ${{ github.ref == 'refs/heads/master' }}
```
Comment (ryan-roemer) [inline .github/workflows/main.yml:47]: We shouldn't need this.

File: .github/workflows/main.yml [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+        run: |
+          git config --global core.autocrlf false
+      - name: Checkout
+        uses: actions/checkout@v2
+      - name: Setup node
+        uses: actions/setup-node@v2
+        with:
+          node-version: ${{ matrix.node_version }}
+      - name: Test
+        run: |
```
Comment (ryan-roemer) [inline .github/workflows/main.yml:37]: Do we need:

```yaml
if: ${{ matrix.os != 'windows-latest' }}
```

here?

**Comment by ryan-roemer** [comment]
Looking at https://github.com/FormidableLabs/react-fast-compare/blob/master/package.json#L18-L19 

- `yarn test` does a whole bunch of checks then runs `yarn test-browser` for Karma on headless chrome. This works in Mac + Linux + Windows
- `yarn test-ie` does all the same checks except swaps `yarn test-browser` with `yarn test-browser-ie` which does Karma with IE. This only works on Windows.

On windows, we're now doing lots of the exact same checks. We can either:

1. switch to only doing `yarn test` on Mac + linux and `yarn test-ie` on Windows, or
2. Always do `yarn test` on all os's and replace `yarn test-ie` with `yarn test-browser-ie` only on windows.

File: .github/workflows/main.yml (hunk 1 above)
Comment (cdierkens) [reply .github/workflows/main.yml:7]: Good catch!

File: .github/workflows/main.yml (hunk 2 above)
Comment (cdierkens) [reply .github/workflows/main.yml:47]: We don't need to run the `codecov` command or we are safe to run it on both PRs and the master branch?

**Comment by cdierkens** [comment]
> Looking at https://github.com/FormidableLabs/react-fast-compare/blob/master/package.json#L18-L19 
> 
> 
> 
> - `yarn test` does a whole bunch of checks then runs `yarn test-browser` for Karma on headless chrome. This works in Mac + Linux + Windows
> 
> - `yarn test-ie` does all the same checks except swaps `yarn test-browser` with `yarn test-browser-ie` which does Karma with IE. This only works on Windows.
> 
> 
> 
> We can either:
> 
> 1. switch to only doing `yarn test` on Mac + linux and `yarn test-ie` on Windows, or
> 
> 2. Always do `yarn test` on all os's and replace `yarn test-ie` with `yarn test-browser-ie` only on windows.

Do we have a strong preference here? If not I'll number '1'

File: .github/workflows/main.yml (hunk 2 above)
Comment (ryan-roemer) [reply .github/workflows/main.yml:47]: We want to run it in both PRs and master branch. Codecov updates PRs with branch results of +/- coverage and master branch runs give us our cool badge.

File: .github/workflows/main.yml (hunk 3 above)
Comment (cdierkens) [reply .github/workflows/main.yml:37]: That would satisfy the direction of doing "switch to only doing yarn test on Mac + linux and yarn test-ie on Windows". I'll put that in.

**Comment by ryan-roemer** [comment]
@cdierkens 

For

> Do we have a strong preference here? If not I'll switch to number '1'.

Maybe do option 2 because that means we can now also test headless chrome on Windows? Not a biggie either way...

File: .github/workflows/main.yml [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+        uses: actions/checkout@v2
+      - name: Setup node
+        uses: actions/setup-node@v2
+        with:
+          node-version: 14
+      - name: Test
+        run: |
+          yarn install --frozen-lockfile
+          yarn test
+          yarn codecov
```
Comment (ryan-roemer) [inline .github/workflows/main.yml:60]: This is likely going to need a token in secrets. See, e.g. https://github.com/FormidableLabs/trace-deps/blob/main/.github/workflows/ci.yml#L69-L72

I will create and add the secret `CODECOV_TOKEN` in this project.

File: .github/workflows/main.yml [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+      - name: Setup node
+        uses: actions/setup-node@v2
+        with:
+          node-version: 14
+      - name: Test
+        run: |
+          yarn install --frozen-lockfile
```
Comment (ryan-roemer) [inline .github/workflows/main.yml:58]: Please separate out yarn install, test and coverage steps.

File: .github/workflows/main.yml [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+      - name: Checkout
+        uses: actions/checkout@v2
+      - name: Setup node
+        uses: actions/setup-node@v2
+        with:
+          node-version: ${{ matrix.node_version }}
+      - name: Test
+        if: ${{ matrix.os != 'windows-latest' }}
+        run: |
+          yarn install --frozen-lockfile
```
Comment (ryan-roemer) [inline .github/workflows/main.yml:39]: Please separate out yarn install into separate step (and below)

File: .github/workflows/main.yml (hunk 4 above)
Comment (ryan-roemer) [reply .github/workflows/main.yml:60]: UPDATE -- Looks like codecov has updated features. Let's see if things work without it!

**Review by ryan-roemer — APPROVED** [review]
🎉

**Pull request merged**